### PR TITLE
Handle macOS backtrace symbol format

### PR DESCRIPTION
**Pull request by Theodus**
Unexpected symbol format would cause assertions on macOS to hang the process indefinitely rather than dump the flight recorder.

### CODE REVIEW

File: src/rt/test/systematic.h [hunk 1]
```diff
@@ -489,9 +489,49 @@ namespace Systematic
...
+        }
+      }
+      auto* mod_start = strchr(sym, ' ') + 1;
+      auto* addr_start = strchr(mod_start, ' ') + 1;
+      auto* name_start = strchr(addr_start, ' ') + 1;
+      auto* offset_start = strrchr(sym, ' ') + 1;
+
+      auto buf_write = [](char* dst, char const* src, size_t n) -> char* {
+        strncpy(dst, src, n);
+        return dst + n;
```
Comment (davidchisnall) [inline src/rt/test/systematic.h:519]: Why not just pass dst by reference?

**Comment by Theodus** [comment]
This has been updated to just replace the mangled symbol name, using the format of the platform.

**Comment by Theodus** [comment]
The offset is still present:
macOS:
 ```
6   func-con-diningphilosophers         0x0000000102fee33c (void verona::rt::SchedulerThread<verona::rt::Cown>::run<>(void (*)()) + 1084
```
Linux:
```
./src/rt/func-con-diningphilosophers(void verona::rt::SchedulerThread<verona::rt::Cown>::run<>(void (*)())+0x41b) [0x55c8012e7cbb]
```

**Comment by davidchisnall** [comment]
Ah, I see.  I misread the code the first time.  Nice!

**Pull request merged**